app/routes/planet_routes.py

```diff
@@ -0,0 +1,93 @@
+from app import db
+from app.models.planet import Planet
+from flask import Blueprint, jsonify, make_response, request
+from app.routes.routes_helper import error_message
+
+planets_bp = Blueprint("planets", __name__, url_prefix="/planets")
+
+def make_planet_safely(data_dict):
+	try:
+		return Planet.from_dict(data_dict)
+	except KeyError as err:
+		error_message(f"Missing key: {err}", 400)
+
+def replace_planet_safely(planet, data_dict):
+	try:
+		return planet.replace_details(data_dict)
+	except KeyError as err:
+		error_message(f"Missing key: {err}", 400)
+
+@planets_bp.route("", methods=["POST"])
+def handle_planets():
+	request_body = request.get_json()
+	new_planet = make_planet_safely(request_body)
+
+	db.session.add(new_planet)
+	db.session.commit()
+
+	return jsonify(new_planet.to_dict()), 201
+
+@planets_bp.route("", methods=["GET"])
+def index_planets(): 
+	name_param = request.args.get("name")
+
+	if name_param:
+		planets = Planet.query.filter_by(name=name_param)
+	else:
+		planets = Planet.query.all()
```

Comment:
Consider adding a `@classmethod` to `Planet` that could be used to perform this filtering. 

Also, think about how you might filter on multiple fields, or use partial matches. Check out https://docs.sqlalchemy.org/en/13/orm/tutorial.html#common-filter-operators for some ideas!